classes/controller/FrontController.php

```diff
@@ -1825,6 +1826,64 @@ public function getTemplateVarPage()
         return $page;
     }
 
+    /**
+     * Returns structured data for this page. This parent method renders
+     * the basic data for the shop. Other controllers (product, cms) should
+     * further enrich it.
+     *
+     * @return array
+     */
+    public function getStructuredData()
+    {
+        // Basic organization and webpage
+        $structuredData = [
+            'organization' => [
+                '@context' => 'https://schema.org',
+                '@type' => 'Organization',
+                'name' => Configuration::get('PS_SHOP_NAME'),
+                'url' => $this->getTemplateVarUrls()['pages']['index'],
+                'logo' => [
+                    '@type' => 'ImageObject',
+                    'url' => $this->getShopLogo()['src']
+                ],
+            ],
+            'webpage' => [
+                '@context' => 'https://schema.org',
+                '@type' => 'WebPage',
+                'isPartOf' => [
+                    '@type' => 'WebSite',
+                    'url' => $this->getTemplateVarUrls()['pages']['index'],
+                    'name' => Configuration::get('PS_SHOP_NAME'),
+                ],
+                'name' => $this->getTemplateVarPage()['meta']['title'],
+                'url' => $this->getTemplateVarUrls()['current_url'],
+            ],
+        ];
+
+        // Add breadcrumbs
+        $breadcrumbLinks = $this->getBreadcrumbLinks();
+        if (!empty($breadcrumbLinks['links'][0])) {
+            $breadcrumbStructuredData = [
+                '@context' => 'https://schema.org',
+                '@type' => 'BreadcrumbList',
+                'itemListElement' => [],
+            ];
+            $counter = 1;
+            foreach ($breadcrumbLinks['links'] as $link) {
+                $breadcrumbStructuredData['itemListElement'][] = [
+                    '@type' => 'ListItem',
+                    'position' => $counter,
+                    'name' => $link['title'],
+                    'item' => $link['url'],
+                ];
+                $counter++;
+            }
+            $structuredData['breadcrumbs'] = $breadcrumbStructuredData;
+        }
+
+        return $structuredData;
```

Comment:
@Hlavtox I think we can add new hook here to able to modify this data ?

Reply:
No need, there is already an universal hook. :-)

Everyone can modify the seo strategy easily, working with array, using actionFrontControllerSetVariables hook.

Review or comments module can easily add their own data using actionFrontControllerSetVariables hook.